### PR TITLE
[bitnami/postgres-ha] Extra volumes, mount and initContainers added.

### PR DESCRIPTION
**Pull request by lpaarup**
**Description of the change**

Added extraVolumes, extraVolumeMounts and extraInitContainers to the postgresql pods.

**Benefits**

In reference to #3437, the extraVolumes and extraVolumeMounts allow to mount an external volume to the directory where postgres writes the WAL files. This is to have the WAL archiving separate from the main data segment. In this case, if we lose the data segment, we still have the WAL archives in a separate location.

The extraInitContainers are useful in order to give the postgres user the permission to write into the volume.

**Checklist** 
- [X] Support for extraVolumes.
- [X] Support for extraVolumeMounts.
- [X] Support for extraInitContainers.

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 1]
```diff
@@ -71,6 +73,11 @@ spec:
           volumeMounts:
             - name: data
               mountPath: {{ .Values.persistence.mountPath }}
+      {{- end }}
+      {{- if .Values.postgresql.extraInitContainers }}
+      {{- include "common.tplvalues.render" ( dict "value" .Values.postgresql.extraInitContainers "context" $ ) | nindent 8 }}
+      {{- end }}
+
```
Comment (carrodher) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:80]: I would remove this line
```suggestion

```

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 2]
```diff
@@ -51,8 +51,10 @@ spec:
       {{- if .Values.postgresql.securityContext.enabled }}
       securityContext: {{- omit .Values.postgresql.securityContext "enabled" | toYaml | nindent 8 }}
       {{- end }}
-      {{- if and .Values.volumePermissions.enabled .Values.persistence.enabled }}
+
```
Comment (carrodher) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:54]: I would remove this line
```suggestion

```

File: bitnami/postgresql-ha/Chart.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: postgresql-ha
-version: 3.8.1
+version: 3.8.2
```
Comment (carrodher) [inline bitnami/postgresql-ha/Chart.yaml:3]: As it's a new feature addition, it's better to bump the minor version 
```suggestion
version: 3.9.0
```

**Review by carrodher — COMMENTED** [review]
Thanks for your contribution, can you take a look at my comments?

**Comment by lpaarup** [comment]
Hi @carrodher, thanks for the input. I've already updated it.

**Comment by carrodher** [comment]
Thanks, I just launched the PR in our internal testing pipeline. I will update this PR with more information about the result if needed

**Pull request merged**